src/Ssl/Generator/KeyPairGenerator.php

```diff
@@ -24,58 +26,47 @@
  */
 class KeyPairGenerator
 {
+    private $generator;
+
+    public function __construct(PrivateKeyGeneratorInterface $generator = null)
+    {
+        $this->generator = $generator ?: new ChainPrivateKeyGenerator(
+            [
+                new RsaKeyGenerator(),
+                new EcKeyGenerator(),
+            ]
+        );
+    }
+
     /**
      * Generate KeyPair.
      *
-     * @param int $keySize size of the key
+     * @param KeyOption $keyOption configuration of the key to generate
      *
      * @throws KeyPairGenerationException when OpenSSL failed to generate keys
      *
      * @return KeyPair
      */
-    public function generateKeyPair($keySize = 4096)
+    public function generateKeyPair($keyOption = null)
     {
-        Assert::integer($keySize, __METHOD__.'::$keySize should be an integer. Got: %s');
-
-        $key = openssl_pkey_new(
-            [
-                'private_key_type' => OPENSSL_KEYTYPE_RSA,
-                'private_key_bits' => $keySize,
-            ]
-        );
-
-        if (!$key) {
-            throw new KeyPairGenerationException(
-                sprintf(
-                    'OpenSSL key creation failed during generation with error: %s',
-                    openssl_error_string()
-                )
-            );
+        if (null === $keyOption) {
+            $keyOption = new RsaKeyOption();
         }
-
-        if (!openssl_pkey_export($key, $privateKey)) {
-            throw new KeyPairGenerationException(
-                sprintf(
-                    'OpenSSL key export failed during generation with error: %s',
-                    openssl_error_string()
-                )
-            );
+        if (\is_int($keyOption)) {
+            @trigger_error('Passing a keySize to "generateKeyPair" is deprecated since version 1.0 and will be removed in 2.0. Pass an instance of KeyOption instead', E_USER_DEPRECATED);
```

Comment:
Could you put 1.1 there instead (and same for other places)? I will publish 1.1 with this in it :) .